core/src/main/java/de/danoeh/antennapod/core/util/download/AutoUpdateManager.java

```diff
@@ -0,0 +1,113 @@
+        JobScheduler jobScheduler = context.getSystemService(JobScheduler.class);
+        if (jobScheduler != null) {
+            JobInfo oldJob = jobScheduler.getPendingJob(JOB_ID_FEED_UPDATE);
+            if (oldJob == null || oldJob.getIntervalMillis() != intervalMillis) {
+                JobInfo.Builder builder = getFeedUpdateJobBuilder(context);
+                builder.setPeriodic(intervalMillis);
+                jobScheduler.cancel(JOB_ID_FEED_UPDATE);
+
+                if (intervalMillis <= 0) {
+                    Log.d(TAG, "Automatic update was deactivated");
+                    return;
+                }
+
+                jobScheduler.schedule(builder.build());
+                Log.d(TAG, "JobScheduler was set at interval " + intervalMillis);
+            } else {
+                Log.d(TAG, "JobScheduler was already set at interval " + intervalMillis + ", ignoring.");
+            }
+        }
+    }
+
+    public static void restartAlarmManagerInterval(Context context, long triggerAtMillis, long intervalMillis) {
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+
+        if (alarmManager == null) {
+            Log.d(TAG, "AlarmManager was null");
+            return;
+        }
+
+        Intent intent = new Intent(context, FeedUpdateReceiver.class);
+        PendingIntent updateIntent = PendingIntent.getBroadcast(context, 0, intent, 0);
+        alarmManager.cancel(updateIntent);
+
+        if (intervalMillis <= 0) {
+            Log.d(TAG, "Automatic update was deactivated");
+            return;
+        }
+
+        alarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP,
+                SystemClock.elapsedRealtime() + triggerAtMillis,
+                updateIntent);
+        Log.d(TAG, "Changed alarm to new interval " + TimeUnit.MILLISECONDS.toHours(intervalMillis) + " h");
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    public static void restartJobServiceTriggerAt(Context context, long triggerAtMillis) {
+        JobInfo.Builder builder = getFeedUpdateJobBuilder(context);
+        builder.setMinimumLatency(triggerAtMillis);
+        JobScheduler jobScheduler = context.getSystemService(JobScheduler.class);
+        if (jobScheduler != null) {
+            jobScheduler.cancel(JOB_ID_FEED_UPDATE);
+            jobScheduler.schedule(builder.build());
+            Log.d(TAG, "JobScheduler was set for " + triggerAtMillis);
+        }
+    }
+
+    public static void restartAlarmManagerTimeOfDay(Context context, Calendar alarm) {
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        PendingIntent updateIntent = PendingIntent.getBroadcast(context, 0,
+                new Intent(context, FeedUpdateReceiver.class), 0);
+
+        if (alarmManager == null) {
+            Log.d(TAG, "AlarmManager was null");
+            return;
+        }
+
+        alarmManager.cancel(updateIntent);
```

Comment:
this shares a lot of code with `restartAlarmManagerInterval`
we should refactor this further, maybe not in this PR